### PR TITLE
feat: assorted enchancements

### LINKED ISSUE #53: Assorted UI/UX improvements
To be determined/finalized. Some of the points below are more so brainstorming ideas rather than actual requirements to implement.

### Hide or shrink "Ways to experience CUSEC" icons on smallest breakpoint

image

### _Consider_ adding underline highlights to keywords

Like on the [FireFox](https://www.mozilla.org/en-US/firefox/new/) website but with our colors

image

```javascript
import { ReactElement } from "react";
import styled from "@emotion/styled";

const GradientUnderlineSpan = styled.span`
  display: inline-block;
  position: relative;

  &::after {
    background-image: url("https://www.mozilla.org/media/protocol/img/zaps/zap-09.bc0c0b72fdaf.svg");
    background-repeat: no-repeat;
    background-size: 100% 100%;
    content: "";
    left: -0.2em;
    position: absolute;
    top: calc(100% - 0.4em);
    height: 0.3em;
    width: calc(100% + 0.4em);
  }
`;

export default function GradientUnderline(props: {
  children: string | ReactElement;
}): ReactElement {
  return (
    <>
      <GradientUnderlineSpan>{props.children}</GradientUnderlineSpan>
    </>
  );
}
```

### Set a max width container

Limit content to some maximum width to provide a usable experience even on absurdly high resolution displays.

### _Consider_ adding more spacing to nav bar items

They're too close together?

image

Also _consider_ porting the link hover effect from the footer. Something like this:

image

```javascript
export const NavBarLink = styled(Text)`
  font-family: "Inter", sans-serif;
  font-weight: 500;
  margin-right: 32px;
  position: relative;

  &:after {
    background: ${theme.colors.brand.dark_teal};
    content: "";
    position: absolute;
    bottom: -2px;
    left: 0;
    width: 100%;
    height: 4px;
    opacity: 0;
    transform: translateY(3px);
    transition: opacity 0.2s ease, transform 0.2s ease;
  }

  &:hover::after {
    opacity: 1;
    transform: translateY(0);
  }
`;
```

### Make "Ways to experience CUSEC" cards slightly thinner on desktop

The cards can sometimes show up as a single vertical column instead of 2x2 depending on browser/system? Make smaller to accommodate a larger range of displays.

### Adjust min-heights in hero to accommodate the extra space the nav bar takes up

There's not much space at the top especially on something like a 1080p phone with the mobile address bar down.

### _Consider_ tweaking sizing of headlines and section spacing

The spacing between sections feels a bit cramped? Please see Figma for latest mock-up.

### Update hero for landing page (instead of splash page)

- include a register button (replace the email listing signup?)
- consider using "Virtual Event" instead of "Virtual Experience" to be more clear?

### PR DESCRIPTION
**Pull request by sttic**
Resolves #53 <sub><sub>(*not all mentioned points were considered)</sub></sub>

Please visit https://github.com/cusec/2021/pull/62#issuecomment-716826172 or [this link](https://cusec-2021-staging--pr62-feat-assorted-enhanc-ps4e8jt5.web.app) for a preview of the PR.

A bunch of different changes are packed into the PR, mostly dealing with refactoring and style tweaks. The major ones are:

- adopting common components between sections in [components/core/](https://github.com/cusec/2021/tree/feat/assorted-enhancements/components/core)
- introduction of a max-width wrapper to limit how wide the page can be and give consistent padding between sections
- tweaking sizing and spacing so headlines are more prominent and the sections don't feel as cramped together

Sorry for the super messy/long diff :sweat_smile:

### CODE REVIEW

File: components/FooterModal.tsx [hunk 1]
```diff
@@ -61,8 +58,7 @@ export default function FooterModal({
         _hover={{}}
         fontSize={["xs", "xs", "xs", "sm"]}
         onClick={handleOnOpen}
-        rel="noopener noreferrer"
-        target="_blank"
+        isExternal
```
Comment (gordonpn) [inline components/FooterModal.tsx:61]: 😯 Did not know about this!

File: components/sections/Footer.tsx [hunk 2]
```diff
@@ -55,20 +55,17 @@ export default function Footer(): ReactElement {
                 _hover={{}}
                 fontSize={["xs", "xs", "xs", "sm"]}
                 href="https://github.com/cusec/2021"
-                rel="noopener noreferrer"
-                target="_blank"
+                isExternal
               >
                 Open Source
               </FooterLink>
               <FooterLink
```
Comment (gordonpn) [inline components/sections/Footer.tsx:62]: Missing href or TBD?

**Review by gordonpn — APPROVED** [review]
Awesome refactoring Tommy! It's really appreciated 😍

The new core component names are great!

File: components/FooterModal.tsx (hunk 1 above)
Comment (sttic) [reply components/FooterModal.tsx:61]: Super useful especially if like me and never remember the `rel="noopener noreferrer"` :joy:

File: components/sections/Footer.tsx (hunk 2 above)
Comment (sttic) [reply components/sections/Footer.tsx:62]: TBD :sweat_smile: I'm thinking it'd be nicer to have all the attributions listed in the modal component you made!

Comment (gordonpn) [reply components/sections/Footer.tsx:62]: That sounds good!

**Pull request merged**